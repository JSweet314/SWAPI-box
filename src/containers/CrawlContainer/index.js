import React, {Component} from 'react';
import {fetchOpeningCrawl} from '../../apiCalls/fetchOpeningCrawl';
import './style.css';

export default class CrawlContainer extends Component {
  constructor() {
    super();
    this.state = {openingCrawl: '', title: '', releaseDate: ''};
  }

  componentDidMount = () => {
    this.findClosestOpeningCrawl();
  }

  findClosestOpeningCrawl = () => {
    const priorCrawl = localStorage.getItem('SWAPI-crawl');
    if (priorCrawl) {
      this.setState({...JSON.parse(priorCrawl)});
    } else {
      this.getOpeningCrawl();
    }
  }

  getOpeningCrawl = () => {
    const randomFilmNumber = Math.floor(Math.random() * 7) + 1;
    fetchOpeningCrawl(randomFilmNumber)
      .then(this.deployNewCrawl)
      .catch(error => alert(error));
  }

  deployNewCrawl = filmData => {
    const {openingCrawl, title, releaseDate} = filmData;
    this.setState({openingCrawl, title, releaseDate}, this.storeOpeningCrawl);
  };

  storeOpeningCrawl = () => 
    localStorage.setItem('SWAPI-crawl', JSON.stringify(this.state))

  render() {
    const {openingCrawl, title, releaseDate} = this.state;
    return (
      <aside className="scrolling-text">
        <div className="crawl">
          <pre>{openingCrawl}</pre>
          <p>{title}</p>
          <p>{releaseDate}</p>
        </div>
        <h3 className="instructions">Select a Category</h3>
      </aside>
    );
  }
}