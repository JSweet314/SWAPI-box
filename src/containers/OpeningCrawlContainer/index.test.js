import React from 'react';
import CrawlContainer from './index';
import { shallow } from 'enzyme';

describe('ScrollingText', () => {
  it('should match a snapshot with empty state', () => {
    const wrapper = shallow(
      <CrawlContainer />, 
      {disableLifecycleMethods: true}
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should match a snapshot with full state', () => {
    const wrapper = shallow(
      <CrawlContainer />,
      { disableLifecycleMethods: true }
    );
    wrapper.setState({
      openingCrawl: "I am placeholder text!",
      title: "A StarWars Movie",
      releaseDate: "2018-03-23"
    });
    expect(wrapper).toMatchSnapshot();
  });
  
  /*eslint-disable max-len*/
  it('should have a method findClosestOpeningCrawl which sets state with localStorage data if available', () => {
  /*eslint-enable max-len*/
    
  });
});