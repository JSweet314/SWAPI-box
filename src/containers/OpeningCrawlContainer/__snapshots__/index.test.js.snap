// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollingText should match a snapshot of an error 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OpeningCrawlContainer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="crawl"
>
        <h3>
                Something went wrong...
        </h3>
        <h3>
                Please select a category or try again at a latter time.
        </h3>
        <p>
                This is not the page you are looking for...
        </p>
</div>,
      "className": "scrolling-text",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            Something went wrong...
</h3>,
          <h3>
            Please select a category or try again at a latter time.
</h3>,
          <p>
            This is not the page you are looking for...
</p>,
        ],
        "className": "crawl",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Something went wrong...",
          },
          "ref": null,
          "rendered": "Something went wrong...",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Please select a category or try again at a latter time.",
          },
          "ref": null,
          "rendered": "Please select a category or try again at a latter time.",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "This is not the page you are looking for...",
          },
          "ref": null,
          "rendered": "This is not the page you are looking for...",
          "type": "p",
        },
      ],
      "type": "div",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="crawl"
>
          <h3>
                    Something went wrong...
          </h3>
          <h3>
                    Please select a category or try again at a latter time.
          </h3>
          <p>
                    This is not the page you are looking for...
          </p>
</div>,
        "className": "scrolling-text",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              Something went wrong...
</h3>,
            <h3>
              Please select a category or try again at a latter time.
</h3>,
            <p>
              This is not the page you are looking for...
</p>,
          ],
          "className": "crawl",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Something went wrong...",
            },
            "ref": null,
            "rendered": "Something went wrong...",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Please select a category or try again at a latter time.",
            },
            "ref": null,
            "rendered": "Please select a category or try again at a latter time.",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "This is not the page you are looking for...",
            },
            "ref": null,
            "rendered": "This is not the page you are looking for...",
            "type": "p",
          },
        ],
        "type": "div",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;

exports[`ScrollingText should match a snapshot with empty state 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OpeningCrawlContainer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="crawl"
>
          <pre />
          <p />
          <p />
</div>,
        <h3
          className="instructions"
>
          Select a Category
</h3>,
      ],
      "className": "scrolling-text",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <pre />,
            <p />,
            <p />,
          ],
          "className": "crawl",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
            },
            "ref": null,
            "rendered": "",
            "type": "pre",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
            },
            "ref": null,
            "rendered": "",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
            },
            "ref": null,
            "rendered": "",
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Select a Category",
          "className": "instructions",
        },
        "ref": null,
        "rendered": "Select a Category",
        "type": "h3",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="crawl"
>
            <pre />
            <p />
            <p />
</div>,
          <h3
            className="instructions"
>
            Select a Category
</h3>,
        ],
        "className": "scrolling-text",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <pre />,
              <p />,
              <p />,
            ],
            "className": "crawl",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
              },
              "ref": null,
              "rendered": "",
              "type": "pre",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
              },
              "ref": null,
              "rendered": "",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
              },
              "ref": null,
              "rendered": "",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Select a Category",
            "className": "instructions",
          },
          "ref": null,
          "rendered": "Select a Category",
          "type": "h3",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;

exports[`ScrollingText should match a snapshot with full state 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OpeningCrawlContainer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="crawl"
>
          <pre>
                    I am placeholder text!
          </pre>
          <p>
                    A StarWars Movie
          </p>
          <p>
                    2018-03-23
          </p>
</div>,
        <h3
          className="instructions"
>
          Select a Category
</h3>,
      ],
      "className": "scrolling-text",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <pre>
              I am placeholder text!
</pre>,
            <p>
              A StarWars Movie
</p>,
            <p>
              2018-03-23
</p>,
          ],
          "className": "crawl",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "I am placeholder text!",
            },
            "ref": null,
            "rendered": "I am placeholder text!",
            "type": "pre",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "A StarWars Movie",
            },
            "ref": null,
            "rendered": "A StarWars Movie",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "2018-03-23",
            },
            "ref": null,
            "rendered": "2018-03-23",
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Select a Category",
          "className": "instructions",
        },
        "ref": null,
        "rendered": "Select a Category",
        "type": "h3",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="crawl"
>
            <pre>
                        I am placeholder text!
            </pre>
            <p>
                        A StarWars Movie
            </p>
            <p>
                        2018-03-23
            </p>
</div>,
          <h3
            className="instructions"
>
            Select a Category
</h3>,
        ],
        "className": "scrolling-text",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <pre>
                I am placeholder text!
</pre>,
              <p>
                A StarWars Movie
</p>,
              <p>
                2018-03-23
</p>,
            ],
            "className": "crawl",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "I am placeholder text!",
              },
              "ref": null,
              "rendered": "I am placeholder text!",
              "type": "pre",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "A StarWars Movie",
              },
              "ref": null,
              "rendered": "A StarWars Movie",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "2018-03-23",
              },
              "ref": null,
              "rendered": "2018-03-23",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Select a Category",
            "className": "instructions",
          },
          "ref": null,
          "rendered": "Select a Category",
          "type": "h3",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
